tests/serializers/test_json.py

```diff
@@ -252,6 +252,24 @@ def test_helpful_error_message_for_many2many_natural2(self):
             for obj in serializers.deserialize('json', test_string, ignore=False):
                 obj.save()
 
+    def test_helpful_error_message_for_many2many_invalid(self):
+        """
+        Invalid many-to-many keys should throw a helpful error message even
+        when the serialized value is invalid (non-iterable in this case).
+        """
+        data_to_serialize = [{
+            'pk': 1,
+            'model': 'serializers.M2MData',
+            'fields': {'data': None},
+        }]
+
+        serialized = json.dumps(data_to_serialize)
+        expected = "(serializers.M2MData:pk=1) field_value was 'None'"
+
+        with self.assertRaisesMessage(DeserializationError, expected):
+            for _ in serializers.deserialize('json', serialized, ignore=False):
+                pass
```

Comment:
IMO we can simplify this with `next()`, i.e.
```python
with self.assertRaisesMessage(DeserializationError, expected):
    next(serializers.deserialize('json', serialized, ignore=False))
```

Reply:
Good idea, thanks.